packages/truffle-debugger/lib/evm/sagas/index.js

```diff
@@ -45,14 +56,33 @@ export function* begin({ address, binary }) {
 }
 
 export function* callstackSaga () {
+  let contexts;
+  let instances;
+
   while (true) {
     yield take(TICK);
-    debug("got TICK");
+
+    // HACK contexts and instances won't be available until after first TICK
```

Comment:
I don't understand the point of this block; I thought we just got a TICK? What's going on here?

Reply:
I originally had this assignment before the `while` loop, because the contexts/instances never change. But rather than stringing in `READY`, I'm just doing that `if` statement because I know if we have just gotten a TICK, then the session is READY

Reply:
Oh!  I get it now.  The hack is not those lines or the if block, but that it's inside the while-loop.  I'd suggest adding a comment to the code with something like what you just said -- e.g., `//only make this assignment the first time through the loop`, or something to that effect.